src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompiledAssemblyUtility.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    public static class CompiledAssemblyUtility
+    {
+        public static Type GetExportedType(Stream assemblyStream, Stream pdbStream)
+        {
+            var assembly =
+#if NET451
+                Assembly.Load(GetBytes(assemblyStream), GetBytes(pdbStream));
+#else
+                System.Runtime.Loader.AssemblyLoadContext.Default.LoadFromStream(assemblyStream, pdbStream);
+#endif
+            return assembly.GetExportedTypes().FirstOrDefault(a => !a.IsNested);
```

Comment:
Actually, I think this is really gross. I don't think this code is really reusable for anything other than what Razor does today (and it would be nice to improve what Razor does today).